### PR TITLE
Remove liquidity balanced one token and custom

### PR DESCRIPTION
**Pull request by marshall-bean**
Remove liquidity:
- balanced (50/50 ratio)
- one token
- custom ratio

### CODE REVIEW

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  Balanced,
+  OneToken,
+  Custom
+}
+
+type LiquidityAmounts = {
+  [key: number]: TokenValue;
+};
+
+export const RemoveLiquidity = ({ well }: RemoveLiquidityProps) => {
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:26]: I've been using this type for all react functional components, it adds some built in react things to the props, like `children`, tho not needed in this case:

```
import { FC } from "src/types";
export const RemoveLiquidity = FC<RemoveLiquidityProps>({ well }) => {
```

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+      setWellTokens(tokens);
+      setAmounts(initialAmounts);
+    }
+  }, [well]);
+
+  useEffect(() => {
+    if (well.lpToken) {
+      let lpTokenWithMetadata = well.lpToken;
+      lpTokenWithMetadata.setMetadata({ logo: images[well.lpToken.symbol] ?? images.DEFAULT });
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:65]: we should have a different default image for our Well LP tokens. cc @sweetredbeans

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    data: customRatioQuote,
+    isLoading: loadingCustomRatioQuote,
+    isError: customRatioQuoteError
+  } = useQuery(["wells", address, removeLiquidityMode, amounts], async () => {
+    if (!atLeastOneAmountNonzero() || removeLiquidityMode !== REMOVE_LIQUIDITY_MODE.Custom) {
+      return null;
+    }
+
+    return well.removeLiquidityImbalancedQuote(Object.values(amounts));
+  });
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:105]: should we put these in a single hook, like 
useRemoveLiquidityQuote(quoteType, amounts[], tokenIndex?), then based on the params, return the correct useQuery result?

Also, for the useQuery cache keys:
- not sure why we need `removeLIquidityMode`
- the amounts used for quote... if we really want to cache the quote result, we should change the default ttls

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+  const [amounts, setAmounts] = useState<LiquidityAmounts>({});
+
+  const { isLoading: isAllTokenLoading, refetch: refetchBalances } = useAllTokensBalance();
+
+  const atLeastOneAmountNonzero = useCallback(() => Object.values(amounts).filter((amount) => amount.value.gt("0")).length > 0, [amounts]);
+
+  useEffect(() => {
+    const fetching = isAllTokenLoading;
+    fetching ? setIsLoadingAllBalances(true) : setTimeout(() => setIsLoadingAllBalances(false), 500);
+  }, [isAllTokenLoading]);
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:45]: can we skip this and let the parent, Wells page, handle it? Maybe not show this component if the tokens are still loading?

**Review by 0xalecks — APPROVED** [review]

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx (hunk 4 above)
Comment (marshall-bean) [reply projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:45]: Good idea, I'll implement that in the next PR.

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx (hunk 3 above)
Comment (marshall-bean) [reply projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:105]: Yes they do belong in a separate hook for sure. I was treating it as a "dependency" as the docs suggested.

> if we really want to cache the quote result, we should change the default ttls
Can you elaborate on this one?

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx (hunk 2 above)
Comment (marshall-bean) [reply projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:65]: I think it's great as is. :trollface:

**Pull request merged**